networkx/algorithms/tests/test_simple_paths.py

```diff
@@ -129,18 +146,34 @@ def test_all_simple_paths_cutoff():
     assert_equal(set(tuple(p) for p in paths), {(0, 1), (0, 2, 1), (0, 3, 1)})
 
 
+def test_all_simple_paths_non_trivial():
```

Comment:
I cannot easily tell from looking at this test whether the test was written correctly. Can you summarize the expected result in the function name?